dbt/adapters/athena/lf_permissions.py

```diff
@@ -0,0 +1,149 @@
+from typing import Dict, List
+
+from mypy_boto3_lakeformation import LakeFormationClient
+from mypy_boto3_lakeformation.type_defs import (
+    BatchPermissionsRequestEntryTypeDef,
+    DataCellsFilterTypeDef,
+)
+from pydantic import BaseModel
+
+from dbt.adapters.athena.relation import AthenaRelation
+from dbt.events import AdapterLogger
+
+logger = AdapterLogger("AthenaLakeFormation")
+
+
+class FilterConfig(BaseModel):
+    row_filter: str
+    column_names: List[str] = []
+    principals: List[str] = []
+
+    def to_api_repr(self, catalog_id: str, database: str, table: str, name: str) -> DataCellsFilterTypeDef:
+        return {
+            "TableCatalogId": catalog_id,
+            "DatabaseName": database,
+            "TableName": table,
+            "Name": name,
+            "RowFilter": {"FilterExpression": self.row_filter},
+            "ColumnNames": self.column_names,
+            "ColumnWildcard": {"ExcludedColumnNames": []},
+        }
+
+    def to_update(self, existing: DataCellsFilterTypeDef) -> bool:
+        return self.row_filter != existing["RowFilter"]["FilterExpression"] or set(self.column_names) != set(
+            existing["ColumnNames"]
+        )
+
+
+class DataCellFiltersConfig(BaseModel):
+    enabled: bool = False
+    filters: Dict[str, FilterConfig]
+
+
+class LfGrantsConfig(BaseModel):
+    data_cell_filters: DataCellFiltersConfig
+
+
+class LfPermissions:
+    def __init__(self, catalog_id: str, relation: AthenaRelation, lf: LakeFormationClient) -> None:
```

Comment:
I know that the name `lf` is appealing but could we maybe call `lf_client` instead?

Reply:
no problem, fixed